### PR TITLE
Implement base Story Translation page

### PR DESCRIPTION
**Pull request by gaoxk**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Implement-dumb-admin-story-translation-page-731159c382c5406ea3ea3fb619465d16)

## Implementation description

- implement side bar 
  - backend needs to be modified to return the translator/reviewer names and the total number of lines
- implemented form 
  - making a dumb non functioning form for now
- implemented bottom footer  
  - buttons are non functioning
- using default nav for now   

tldr, only functioning thing is the delete button

## Steps to test

1. login as angela merkel
2. go to http://localhost:3000/story/1/1
3. modify the last number in the url to see different story translations
4. attempt to delete translation 
5. observe redirect to homepage
6. attempt to open the last url. observe the 404

## What should reviewers focus on?

- does it work?

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/homepage/StoryCard.tsx [hunk 1]
```diff
@@ -47,18 +49,7 @@ const StoryCard = ({
 }: StoryCardProps) => {
   const { authenticatedUser } = useContext(AuthContext);
   const history = useHistory();
-  const embedLink = (originalYoutubeLink: string): string => {
```
Comment (gaoxk) [inline frontend/src/components/homepage/StoryCard.tsx:50]: moved this to Utils.ts

**Comment by jennifer-lu** [comment]
Very cool!

Not sure if this is expected behaviour, but changing the last number in the url only changes the story translation every two numbers (i.e. http://localhost:3000/story/1/1 and http://localhost:3000/story/1/2 both show East of Eden, http://localhost:3000/story/1/3 and http://localhost:3000/story/1/4 both show War and Peace.

Also, deleting a translation redirects me to http://localhost:3000/404, not the homepage.

**Comment by gaoxk** [comment]
regarding url changes: thats expected behavior right now. we might migrate away from story/storyid/storytranslationid to story/storytranslationid if the story id proves to be useless

regarding redirect, ill investigate that now!

**Comment by gaoxk** [comment]
redirect after deleting should work now :)

**Comment by Puepis** [comment]
> 
> 
> Very cool!
> 
> Not sure if this is expected behaviour, but changing the last number in the url only changes the story translation every two numbers (i.e. http://localhost:3000/story/1/1 and http://localhost:3000/story/1/2 both show East of Eden, http://localhost:3000/story/1/3 and http://localhost:3000/story/1/4 both show War and Peace.
> 
> Also, deleting a translation redirects me to http://localhost:3000/404, not the homepage.

`story/1/1` and `story/1/2` actually show different story translations (two different languages), just under the same story - same with /3 and /4

**Comment by jennifer-lu** [comment]
> > Very cool!
> > Not sure if this is expected behaviour, but changing the last number in the url only changes the story translation every two numbers (i.e. http://localhost:3000/story/1/1 and http://localhost:3000/story/1/2 both show East of Eden, http://localhost:3000/story/1/3 and http://localhost:3000/story/1/4 both show War and Peace.
> > Also, deleting a translation redirects me to http://localhost:3000/404, not the homepage.
> 
> `story/1/1` and `story/1/2` actually show different story translations (two different languages), just under the same story - same with /3 and /4

I see, thanks for letting me know.

File: frontend/src/components/pages/ManageStoryTranslationPage.tsx [hunk 2]
```diff
@@ -1,22 +1,241 @@
...
+          </Button>
+        </Flex>
+        <Box
+          css={{
+            borderRadius: "8px",
+            margin: "40px",
+            MozBorderRadius: "8px",
+            overflow: "hidden",
+            webkitBorderRadius: "8px",
+          }}
```
Comment (jennifer-lu) [inline frontend/src/components/pages/ManageStoryTranslationPage.tsx:189]: Nit: for me this box is too tall so the bottom two corners of the iframe are not rounded.

**Review by jennifer-lu — APPROVED** [review]
Looks good to me, everything is working as expected!

**Review by Puepis — COMMENTED** [review]
looks good for the most part! just some nits and the embedded video player looks small on my screen, might have something to do with margin/padding/flex

image

**Review by lynn-kim — CHANGES_REQUESTED** [review]

File: frontend/src/components/pages/ManageStoryTranslationPage.tsx [hunk 3]
```diff
@@ -1,22 +1,241 @@
...
+          </FormControl>
+          <Heading size="sm" marginTop="24px">
+            Delete Story Translatoin
+          </Heading>
+          <Text>
+            Permanently delete this story translation and all data associated
+            with it.
+          </Text>
+          <Button
+            colorScheme="red"
```
[the remark this answers is not in the document]
Comment (gaoxk) [reply frontend/src/components/pages/ManageStoryTranslationPage.tsx:174]: image
image

leaving a todo for correcting the way we use the outline variant in a future pr. currently `outline` is explicitly defined in `Button.ts` and is unfortunately blue. removing the explicitly defined `outline` leads to styling regression in the homepage.

File: frontend/src/components/pages/ManageStoryTranslationPage.tsx [hunk 4]
```diff
@@ -1,22 +1,241 @@
...
-      <h1>
-        Manage story translation {storyIdParam} {storyTranslationIdParam} :)
-      </h1>
-    </Box>
+    <Flex direction="column" height="100vh">
+      <Header title="Manage Story Translation" />
+      <Flex direction="row" flex={1}>
+        <Flex sx={filterStyle} maxWidth="292px">
+          <Heading size="lg">{title}</Heading>
+          <Heading size="sm">Story ID</Heading>
```
[the remark this answers is not in the document]
Comment (gaoxk) [reply frontend/src/components/pages/ManageStoryTranslationPage.tsx:127]: IMO i think this is fine: we're not aligned on most pages with headers 
image

**Review by lynn-kim — APPROVED** [review]

File: frontend/src/components/pages/ManageStoryTranslationPage.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (lynn-kim) [reply frontend/src/components/pages/ManageStoryTranslationPage.tsx:127]: sounds good!

**Pull request merged**